src/ctia/schemas/core.clj

```diff
@@ -20,196 +20,225 @@
              [schema :as f-schema]
              [spec :as f-spec]
              [utils :as fu]]
-            [schema.core :refer [Str Bool] :as sc]))
+            [schema.core :refer [Str Bool] :as sc]
+            [schema-tools.core :as st]))
+
+
+
+(sc/defschema ACLEntity
+  (st/optional-keys
+   {:authorized_users [Str]
+    :authorized_groups [Str]}))
+
+(sc/defschema ACLStoredEntity
+  (st/merge ACLEntity
+            {:groups [Str]}))
 
 (defmacro defschema [name-sym ddl spec-kw-ns]
   `(do
-     (sc/defschema ~name-sym (f-schema/->schema ~ddl))
+     (sc/defschema ~name-sym
+       (f-schema/->schema ~ddl))
      (f-spec/->spec ~ddl ~spec-kw-ns)))
 
+(defmacro def-stored-schema [name-sym ddl spec-kw-ns]
+  `(do
+     (sc/defschema ~name-sym
+       (st/merge ACLStoredEntity
+                 (f-schema/->schema ~ddl)))
+     (f-spec/->spec ~ddl ~spec-kw-ns)))
+
+(defmacro def-acl-schema [name-sym ddl spec-kw-ns]
+  `(do
+     (sc/defschema ~name-sym
+       (st/merge ACLEntity
+                 (f-schema/->schema ~ddl)))
+     (f-spec/->spec ~ddl ~spec-kw-ns)))
```

Comment:
What are the consequences of having the spec and the schema not match?  We are validating new entities with spec now in CTIA.  Do they need to match?  Make sure you merge in master and then run test.

There is a feature in flanders that might help if you want to change the flanders schema definitions before generating plumatic schemas and clojure specs.  You can modified a flanders schema by composing the schema defintion with a declarative list of changes.  See [flanders.munge-test](https://github.com/threatgrid/flanders/blob/master/test/flanders/munge_test.clj) for examples.

Reply:
Actually we started the private-ctia branch long before those improvements, after we merge that, there's going to be a new phase were we merge back with CTIA main branch.
We should fix that at this step because everything will be broken anyway.